rules_gathering/BUILD

```diff
@@ -31,3 +33,17 @@ exports_files(
     ]),
     visibility = ["//doc_build:__pkg__"],
 )
+
+
+toolchain_type(name = "sbom_tools")
```

Comment:
Nit: I would prefer toolchain stuffs to live in a separate dir.

That way the target label is more expressive `@rules_gathering//:sbom_tools` -> `@rules_gathering//tools/sbom:toolchain_type`. This follows the convention of `@bazel_tools//tools/cpp:toolchain_type`.